doozerlib/operator_metadata.py

```diff
@@ -575,6 +575,65 @@ def operator(self):
         return self.runtime.image_map[self.operator_name]
 
 
+class OperatorMetadataLatestNvrReporter:
+    """Query latest operator metadata based on nvr and stream"""
+
+    @log
+    def __init__(self, operator_nvr, stream, runtime):
+        self.operator_nvr = operator_nvr
+        self.stream = stream
+        self.operator_name, self.operator_version = self.unpack_operator_nvr()
+        self.metadata_name = '{}-metadata-container'.format(self.operator_name)
+        self.runtime = runtime
+
+    @log
+    def get_latest_build(self):
+        regex = r'^{}-metadata-container-{}\.{}-(?P<release>[0-9]+)$'.format(
+            re.escape(self.operator_name),
+            re.escape(self.operator_version),
+            self.stream)
+        matcher = re.compile(regex)
+        release = 0
+        candidate = None
+
+        for brew_build in self.get_all_builds():
+            result = matcher.match(brew_build)
+            try:
+                matched_release = int(result.group('release'))
+                if matched_release > release:
+                    release = matched_release
+                    candidate = brew_build
+            except AttributeError:
+                pass
+
+        if not candidate:
+            # XXX: This fallback should be removed after operator metadata containers get built with
+            # the new naming scheme. https://jira.coreos.com/browse/ART-1175
+            logger.warning('Did not find a match under the new naming scheme. Falling back to the old')
+            candidate = OperatorMetadataLatestBuildReporter(self.operator_name, self.runtime).get_latest_build()
+
+        return candidate
+
+    @log
+    def get_all_builds(self):
+        """Ask brew for all releases of a package"""
+        cmd = 'brew list-builds --package {} --state COMPLETE --quiet'.format(self.metadata_name)
```

Comment:
hmm... this gives you all the package builds ever for any release...
`brew list-tagged rhaos-4.2-rhel-7-candidate  cluster-logging-operator-metadata-container` would give you just the ones that are eligible for 4.2
I could see some situations where what you have might bring up something that's not actually tagged and can't be added to the advisory.

Reply:
That's a better approach. Thanks!